src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/profiler/ProfilerPresenter.java

```diff
@@ -14,171 +14,164 @@
+      globalDisplay_ = globalDisplay;
       
-      // default profiler commands to disabled until we are attached
-      // to a document and view
-      disableAllCommands();
+      binder.bind(commands, this);
+
+      // by default, one can always start profiling
+      enableStoppedCommands();
    }
-   
+
    public void attatch(SourceDocument doc, Display view)
    {
-      // save references to doc and view
-      doc_ = doc;
-      view_ = view;
-      
-      // initialize view
-      ProfilerContents contents = getContents();
-      view_.getPropA().setValue(contents.getPropA());
-      view_.getPropB().setValue(contents.getPropB());
-      
-      // subscribe to value changes on the view to save contents 
-      // to the server whenenver it's modfied
-      handlerRegistrations_.add(view_.getPropA()
-                  .addValueChangeHandler(new ValueChangeHandler<Integer>() {
-         @Override
-         public void onValueChange(ValueChangeEvent<Integer> event)
-         {
-            contentsUpdater_.nudge();           
-         }
-         
-      }));
-      
-      handlerRegistrations_.add(view_.getPropB()
-                 .addValueChangeHandler(new ValueChangeHandler<Boolean>() {
-         @Override
-         public void onValueChange(ValueChangeEvent<Boolean> event)
-         {
-            contentsUpdater_.nudge();
-         }
-      }));
-      
       // enable commands for stopped state
       enableStoppedCommands();
    }
-   
+
    public void detach()
    {
       // unsubscribe from view
       handlerRegistrations_.removeHandler();
-      
-      // null out references to doc and view
-      doc_ = null;
-      view_ = null;
-      
+
       // disable all commands
       disableAllCommands();
    }
-   
+
    @Handler
    public void onStartProfiler()
    {
-     
-      // manage commands
-      enableStartedCommands();
+      dependencyManager_.withProfvis(profilerDependecyUserAction_, new Command()
+      {
+         @Override
+         public void execute()
+         {
+            // manage commands
+            enableStartedCommands();
+
+            ProfileOperationRequest request = ProfileOperationRequest
+                  .create("");
+            server_.startProfiling(request,
+                  new ServerRequestCallback<ProfileOperationResponse>()
+            {
+               @Override
+               public void onResponseReceived(ProfileOperationResponse response)
+               {
+                  if (response.getErrorMessage() != null)
+                  {
+                     globalDisplay_.showErrorMessage("Profiler Error",
+                           response.getErrorMessage());
+                     return;
+                  }
+
+                  response_ = response;
+               }
+
+               @Override
+               public void onError(ServerError error)
+               {
+                  globalDisplay_.showErrorMessage("Failed to Stop Profiler",
```

Comment:
Assuming "we" as in the IDE, then yes. Made a change to be able to stop profiling if you trigger the `rprof` from console.